bin/lint-package-json

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+set -euf -o pipefail
+
+# shellcheck source=functions
+source "${BASH_SOURCE%/*}/../sanctuary-scripts/functions"
+
+run_custom_script lint-package-json "$@"
+
+owner="$(get repo-owner)"
+name="$(get repo-name)"
+
+run lint-json package.json
+
+declare -i failures=0
+
+assert() {
+  if node <<EOF
+    var pkg = require('./package.json');
+
+    var isArray = Array.isArray;
+
+    function isSortedObject(x) {
+      return typeof x == 'object' && x != null &&
+             (JSON.stringify(Object.keys(x)) ===
+              JSON.stringify(Object.keys(x).sort()));
+    }
+
+    function matches(x, pattern) {
+      return typeof x == 'string' && pattern.test(x);
+    }
+
+    process.exit(($1) === true ? 0 : 1);
+EOF
+  then
+    pass "$1"
+  else
+    fail "$1"
+    (( failures += 1 ))
+  fi
+}
+
+assert "pkg.name === '$name' || pkg.name === '@$owner/$name'"
+
+assert "matches(pkg.version, /^[0-9]+[.][0-9]+[.][0-9]+$/)"
```

Comment:
:warning: This will fail for pre-release versions. We should either remove this assertion or relax the pattern.

Reply:
I used `/^\d+[.]\d+[.]\d+(?:-[a-z0-9.]+)?(?:[+][a-z0-9.]+)?$/` in one of my own projects.

Reply:
Let's just use `/./` here. All I really want to ensure is that the field is present.

Reply:
The reason I've used such specific checking, is because another part of my code relies on this value having that specific format.

The same might be said about [`bin/release`](https://github.com/sanctuary-js/sanctuary-scripts/blob/davidchambers/everything/bin/release) depending on the format of `pkg.version`, so I think it's worth considering the strict validation so to prevent `release` from failing in an unexpected way.

Reply:
`release` would fail with a message such as `Invalid version: XXX` before performing any irreversible actions, so I don't think extra validation is warranted. Furthermore, one should rarely need to edit `pkg.version` manually so invalid values are unlikely.